src/sycl/comm/fp8_descale.h

```diff
@@ -0,0 +1,132 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ **************************************************************************************************/
+
+#pragma once
+
+#include <cutlass/bfloat16.h>
+#include <cutlass/numeric_conversion.h>
+
+#include <cute/layout.hpp>
+#include <cute/numeric/numeric_types.hpp>
+#include <cute/pointer.hpp>
+#include <cute/tensor_impl.hpp>
+#include <cute/underscore.hpp>
+#include <cute/util/sycl_vec.hpp>
+#include <cutlass/detail/helper_macros.hpp>
+
+// Helper device function for E4M3 -> BFLOAT16 bitwise conversion
+CUTLASS_DEVICE uint16_t fp8_e4m3_to_bf16_bitwise(uint8_t const& src) {
+  // E4M3 (1-4-3) constants
+  constexpr uint32_t e4m3_exp_bias = 7;
+  // BFLOAT16 (1-8-7) constants
+  constexpr uint32_t bf16_exp_bias = 127;
+
+  // Unpack FP8 bits
+  uint16_t sign = static_cast<uint16_t>(src & 0x80);
+  uint16_t exponent = static_cast<uint16_t>(src & 0x78) >> 3;
+  uint16_t mantissa = static_cast<uint16_t>(src & 0x07);
+
+  // Reconstruct BFLOAT16 bits
+  uint16_t bf16_sign = sign << 8;
+  // Re-bias exponent and shift to BFLOAT16 position
+  uint16_t bf16_exponent = (exponent - e4m3_exp_bias + bf16_exp_bias) << 7;
+  // Shift mantissa to BFLOAT16 position
+  uint16_t bf16_mantissa = mantissa << 4;
+
+  return bf16_sign | bf16_exponent | bf16_mantissa;
+}
+
+// Helper device function for E5M2 -> BFLOAT16 bitwise conversion
+CUTLASS_DEVICE uint16_t fp8_e5m2_to_bf16_bitwise(uint8_t const& src) {
+  // E5M2 (1-5-2) constants
+  constexpr uint32_t e5m2_exp_bias = 15;
+  // BFLOAT16 (1-8-7) constants
+  constexpr uint32_t bf16_exp_bias = 127;
+
+  // Unpack FP8 bits
+  uint16_t sign = static_cast<uint16_t>(src & 0x80);
+  uint16_t exponent = static_cast<uint16_t>(src & 0x7C) >> 2;
+  uint16_t mantissa = static_cast<uint16_t>(src & 0x03);
+
+  // Reconstruct BFLOAT16 bits
+  uint16_t bf16_sign = sign << 8;
+  // Re-bias exponent and shift to BFLOAT16 position
+  uint16_t bf16_exponent = (exponent - e5m2_exp_bias + bf16_exp_bias) << 7;
+  // Shift mantissa to BFLOAT16 position
+  uint16_t bf16_mantissa = mantissa << 5;
+
+  return bf16_sign | bf16_exponent | bf16_mantissa;
+}
+
+template <typename Encoding, int VectorizeSize = 8, typename SrcTensor, typename DstTensor>
+CUTLASS_DEVICE void convert_and_descale(SrcTensor const& src, DstTensor& dst, float scale) {
+  using SrcVec_u8 = sycl::vec<uint8_t, VectorizeSize>;
+  using DstVec_u16 = sycl::vec<uint16_t, VectorizeSize>;
```

Comment:
are the dtypes of src and dst fixed as uint8_t and uint16_t? If yes, we may refine `typename SrcTensor` and `typename DstTensor` which do not contain info about uint8 and uint16.